frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/vm-settings-tab/os-flavor.tsx

```diff
@@ -72,45 +72,48 @@ export const OSFlavor: React.FC<OSFlavorProps> = React.memo(
     const displayOnly = !!display;
     const cloneBaseDiskImage = iGetFieldValue(cloneBaseDiskImageField);
     const mountWindowsGuestTools = iGetFieldValue(mountWindowsGuestToolsField);
+    const isUserTemplateValid = iGetIsLoaded(iUserTemplate) && !iGetLoadError(iUserTemplate);
+
     const params = {
-      userTemplate,
       flavor,
       workload: workloadProfile,
       os,
     };
 
-    const vanillaTemplates = immutableListToShallowJS(
-      concatImmutableLists(iGetLoadedData(commonTemplates), iGetLoadedData(userTemplates)),
-    );
+    const templates = isUserTemplateValid
+      ? [toShallowJS(iGetLoadedData(iUserTemplate))]
+      : immutableListToShallowJS(iGetLoadedData(commonTemplates));
 
     let operatingSystems;
 
     if (displayOnly) {
       operatingSystems = [{ name: display, id: display }];
     } else {
       operatingSystems = openshiftFlag
-        ? ignoreCaseSort(getOperatingSystems(vanillaTemplates, params.userTemplate), ['name'])
+        ? ignoreCaseSort(getTemplateOperatingSystems(templates), ['name'])
         : operatingSystemsNative;
     }
 
-    const flavors = flavorSort(getFlavors(vanillaTemplates, params));
+    const flavors = flavorSort(getFlavors(templates, params));
 
-    const workloadProfiles = getWorkloadProfiles(vanillaTemplates, params);
+    const workloadProfiles = getWorkloadProfiles(templates, params);
 
     const loadingResources = openshiftFlag
       ? {
-          userTemplates,
           commonTemplates,
           cnvBaseImages,
         }
       : {};
 
+    if (openshiftFlag && iUserTemplate) {
+      Object.assign(loadingResources, { [iUserTemplate]: iUserTemplate });
```

Comment:
```suggestion
      Object.assign(loadingResources, { iUserTemplate: iUserTemplate });
```
the key is just called iUserTemplate (this notation would resolve in a key called `[object Object]`)